content/en/docs/tasks/traffic-management/locality-load-balancing/before-you-begin/index.md

```diff
@@ -57,49 +59,23 @@ metadata:
 EOF
 {{< /text >}}
 
-Add the `sample` namespace to each cluster:
+Next, add the `sample` namespace to each cluster, generate and deploy 
+the `HelloWorld` YAML for each locality, using the locality or localities 
+found in each cluster as the version string:
 
 {{< text bash >}}
-$ for CTX in "$CTX_PRIMARY" "$CTX_R1_Z1" "$CTX_R1_Z2" "$CTX_R2_Z3" "$CTX_R3_Z4"; \
+$ for CTX in ${CTX_PRIMARY} ${CTX_WORKERS[@]}; \
   do \
     kubectl --context="$CTX" apply -f sample.yaml; \
+    for LOC in $(kubectl --context="$CTX" get nodes -o \
+      custom-columns=REGION:'{.metadata.labels.topology\.kubernetes\.io/region}',ZONE:'{.metadata.labels.topology\.kubernetes\.io/zone}' --no-headers  \
+      | uniq | sed -E 's/<none>/default/g' | sed -E 's/[ \t]+/\./g')
+    do \
+      ./@samples/helloworld/gen-helloworld.sh \
+        --version "$LOC" > "helloworld-${LOC}.yaml"; \
+        kubectl --context="$CTX" apply -f "helloworld-${LOC}.yaml"; \
+    done
```

Comment:
This is terse and clever, but I'm afraid this make it less readable and more difficult to modify. Open to other opinions

Reply:
I take that as a compliment :)
Of course you‘re right, but my point here is that I think it should be clear that we have to use the actual region & zone labels from the clusters in Istio to make load balancing work. I can split that commands, no problem...

related (different repo):
https://github.com/istio/istio/issues/30832

Reply:
+1 ... we need to prioritize readability here.